### PR TITLE
Add frequency monitor to hipBLASlt - bench

### PR DESCRIPTION
**Pull request by pkamd**
This PR ports the frequency monitor present in rocBLAS to hipBLASlt that allows reporting frequency information while using hipBLASlt-bench. 

2 flags have been added to enable this feature - 

- ```HIPBLASLT_BENCH_FREQ=true``` to report minimum of the (average and median) XCD frequencies. 
- ```HIPBLASLT_BENCH_FREQ_ALL=true``` to report all of the (average and median) XCD frequencies.

Example usage 
- ```HIPBLASLT_BENCH_FREQ_ALL=true ./staging/hipblaslt-bench -f matmul -r s -m 4000 -n 4000 -k 4000 --lda 4000 --ldb 4000 --ldc 4000 --transA N --transB T -i 10000 -j 1000```
- ```HIPBLASLT_BENCH_FREQ=true ./staging/hipblaslt-bench -f matmul -r s -m 4000 -n 4000 -k 4000 --lda 4000 --ldb 4000 --ldc 4000 --transA N --transB T -i 10000 -j 1000```

### CODE REVIEW

**Review by KKyang — APPROVED** [review]

**Review by babakpst — APPROVED** [review]
@pkamd please merge the PR, once code owners approve the changes.

**Review by KKyang — CHANGES_REQUESTED** [review]
Can you use rebase instead of merge develop into this? If there is any reason to do that I'll change to approve again.

**Review by KKyang — APPROVED** [review]
nvm, I fixed it. Will merge/ Feel free to merge when tests pass.

**Pull request merged**
